### ISSUE TITLE
RADIUS/UAM flow not performing radius client authentication (Openwrt)

### ISSUE THREAD

**Issue by PWJW** [f00b4r0/uspot#10]
I have been trying to get the RADIUS and/or UAM flow working for my test setup.

I have configured everything as per the docs but am at a loss as to why no RADIUS traffic is happening when I enter some credentials on the splash page.

I am using OpenWRT and have built by own firmware (including the uspot and associated dependencies).

Note: If I configure the `auth_mode` as `click-to-continue` it works fine.

Installed packages and versions:

```
base-files - 1588~c2f52e42b1
busybox - 1.36.1-r1
ca-bundle - 20240203-r1
conntrack - 1.4.8-r1
dnsmasq - 2.90-r2
dropbear - 2022.83-r1
firewall4 - 2023.11.03~698a5335-r1
fstools - 2024.01.22~08cd7083-r1
fwtool - 2019.11.12~8f7fe925-r1
getrandom - 2024.01.24~2c5887cb-r2
hostapd-common - 2024.03.09~695277a5-r1
iw - 5.19-r1
iwinfo - 2024.03.23~79a96150-r1
jansson4 - 2.14-r3
jshn - 2024.03.29~eb9bcb64-r1
jsonfilter - 2024.01.23~594cfa86-r1
kernel - 6.1.82~de32b016235a78f4bbc32822a973808e-r1
kmod-ath - 6.1.82.6.6.15-r1
kmod-ath9k - 6.1.82.6.6.15-r1
kmod-ath9k-common - 6.1.82.6.6.15-r1
kmod-cfg80211 - 6.1.82.6.6.15-r1
kmod-crypto-aead - 6.1.82-r1
kmod-crypto-ccm - 6.1.82-r1
kmod-crypto-cmac - 6.1.82-r1
kmod-crypto-crc32c - 6.1.82-r1
kmod-crypto-ctr - 6.1.82-r1
kmod-crypto-gcm - 6.1.82-r1
kmod-crypto-gf128 - 6.1.82-r1
kmod-crypto-ghash - 6.1.82-r1
kmod-crypto-hash - 6.1.82-r1
kmod-crypto-hmac - 6.1.82-r1
kmod-crypto-manager - 6.1.82-r1
kmod-crypto-null - 6.1.82-r1
kmod-crypto-rng - 6.1.82-r1
kmod-crypto-seqiv - 6.1.82-r1
kmod-crypto-sha512 - 6.1.82-r1
kmod-gpio-button-hotplug - 6.1.82-r3
kmod-lib-crc-ccitt - 6.1.82-r1
kmod-lib-crc32c - 6.1.82-r1
kmod-mac80211 - 6.1.82.6.6.15-r1
kmod-nf-conntrack - 6.1.82-r1
kmod-nf-conntrack-netlink - 6.1.82-r1
kmod-nf-conntrack6 - 6.1.82-r1
kmod-nf-flow - 6.1.82-r1
kmod-nf-log - 6.1.82-r1
kmod-nf-log6 - 6.1.82-r1
kmod-nf-nat - 6.1.82-r1
kmod-nf-reject - 6.1.82-r1
kmod-nf-reject6 - 6.1.82-r1
kmod-nfnetlink - 6.1.82-r1
kmod-nft-core - 6.1.82-r1
kmod-nft-fib - 6.1.82-r1
kmod-nft-nat - 6.1.82-r1
kmod-nft-offload - 6.1.82-r1
kmod-nls-base - 6.1.82-r1
kmod-ppp - 6.1.82-r1
kmod-pppoe - 6.1.82-r1
kmod-pppox - 6.1.82-r1
kmod-random-core - 6.1.82-r1
kmod-slhc - 6.1.82-r1
kmod-usb-chipidea - 6.1.82-r1
kmod-usb-chipidea2 - 6.1.82-r1
kmod-usb-core - 6.1.82-r1
kmod-usb-ehci - 6.1.82-r1
kmod-usb-gadget - 6.1.82-r1
kmod-usb-phy-nop - 6.1.82-r1
kmod-usb-roles - 6.1.82-r1
libblobmsg-json20240329 - 2024.03.29~eb9bcb64-r1
libc - 1.2.4-r4
libgcc1 - 13.2.0-r4
libgmp10 - 6.3.0-r1
libiwinfo-data - 2024.03.23~79a96150-r1
libiwinfo20230701 - 2024.03.23~79a96150-r1
libjson-c5 - 0.17-r1
libjson-script20240329 - 2024.03.29~eb9bcb64-r1
liblucihttp-ucode - 2023.03.15~9b5b683f-r1
liblucihttp0 - 2023.03.15~9b5b683f-r1
libmbedtls13 - 2.28.7-r2
libmnl0 - 1.0.5-r1
libnetfilter-conntrack3 - 1.0.9-r2
libnetfilter-cthelper0 - 1.0.0-r2
libnetfilter-cttimeout1 - 1.0.0-r2
libnetfilter-queue1 - 1.0.5-r4
libnettle8 - 3.9.1-r1
libnfnetlink0 - 1.0.2-r1
libnftnl11 - 1.2.6-r1
libnl-tiny1 - 2023.12.05~965c4bf4-r1
libpcap1 - 1.10.4-r1
libpthread - 1.2.4-r4
libradcli - 1.3.0-r1
libubox20240329 - 2024.03.29~eb9bcb64-r1
libubus20231128 - 2023.11.28~f84eb599-r1
libuci20130104 - 2023.08.10~5781664d-r1
libuclient20201210 - 2024.04.05~6c16331e-r1
libucode20230711 - 2024.04.07~5507654a-r1
libudebug - 2023.12.06~6d3f51f9
libustream-mbedtls20201210 - 2024.04.07~d61493a4-r1
logd - 2024.01.24~2c5887cb-r2
mtd - 26
netifd - 2024.01.04~f01345ec-r1
nftables-json - 1.0.9-r1
odhcp6c - 2023.05.12~bcd28363-r20
odhcpd-ipv6only - 2023.10.24~d8118f6e-r1
openwrt-keyring - 2022.03.25~62471e69-r2
opkg - 2022.02.24~d038e5b6-r2
ppp - 2.4.9_git20210104-r5
ppp-mod-pppoe - 2.4.9_git20210104-r5
procd - 2024.03.30~946552a7-r1
procd-seccomp - 2024.03.30~946552a7-r1
procd-ujail - 2024.03.30~946552a7-r1
resolveip - 2
swconfig - 12
tcpdump-mini - 4.99.4-r1
uboot-envtools - 2024.01-r2
ubox - 2024.01.24~2c5887cb-r2
ubus - 2023.11.28~f84eb599-r1
ubusd - 2023.11.28~f84eb599-r1
uci - 2023.08.10~5781664d-r1
uclient-fetch - 2024.04.05~6c16331e-r1
ucode - 2024.04.07~5507654a-r1
ucode-mod-fs - 2024.04.07~5507654a-r1
ucode-mod-log - 2024.04.07~5507654a-r1
ucode-mod-math - 2024.04.07~5507654a-r1
ucode-mod-nl80211 - 2024.04.07~5507654a-r1
ucode-mod-rtnl - 2024.04.07~5507654a-r1
ucode-mod-ubus - 2024.04.07~5507654a-r1
ucode-mod-uci - 2024.04.07~5507654a-r1
ucode-mod-uloop - 2024.04.07~5507654a-r1
uhttpd - 2023.06.25~34a8a74d-r3
uhttpd-mod-ucode - 2023.06.25~34a8a74d-r3
urandom-seed - 3
urngd - 2023.11.01~44365eb1-r1
usign - 2020.05.23~f1f65026-r1
uspot - 2024.03.25~094f0df8-r1
uspot-www - 2024.03.25~094f0df8-r1
uspotfilter - 2024.03.25~094f0df8-r1
wifi-scripts - 1.0-r1
wireless-regdb - 2024.01.23-r1
wpad-basic-mbedtls - 2024.03.09~695277a5-r1
```

Config:

**/etc/config/uspot**

```
config uspot 'captive'
        option auth_mode 'radius'
        option idle_timeout '600'
        option session_timeout '3600'
        option interface 'captive'
        option setname 'uspot'
        option debug '1'

        option auth_server '35.230.xx.xx'
        option auth_port '1812'
        option auth_secret 'xxxxxxxx'
        option nasid 'test'
        option nasmac 'ac:86:74:xx:xx:xx'
        option location_name '111111111'
```

**/etc/config/network**

```
config interface 'loopback'
        option device 'lo'
        option proto 'static'
        option ipaddr '127.0.0.1'
        option netmask '255.0.0.0'

config globals 'globals'
        option ula_prefix 'fdc8:b805:63a5::/48'

config device
        option name 'br-captive'
        option type 'bridge'
        list ports 'phy0-ap0'

config device
        option name 'br-lan'
        option type 'bridge'
        list ports 'eth0'

config interface 'captive'
        option device 'br-captive'
        option proto 'static'
        option ipaddr '10.0.0.1'
        option netmask '255.255.255.0'

config interface 'lan'
        option device 'br-lan'
        option proto 'static'
        option ipaddr '192.168.11.1'
        option netmask '255.255.255.0'
        option ip6assign '60'

config interface 'wan'
        option device 'eth1'
        option proto 'dhcp'

```

**/etc/config/firewall**

```
[default-openwrt-rules-here-snipped]

# create a 'captive' zone for captive portal traffic
config zone
        option name 'captive'
        list network 'captive'
        option input 'REJECT'
        option output 'ACCEPT'
        option forward 'REJECT'

# setup CPD hijacking for unauthenticated clients
config redirect
        option name 'Redirect-unauth-captive-CPD'
        option src 'captive'
        option src_dport '80'
        option proto 'tcp'
        option target 'DNAT'
        option reflection '0'
        option ipset '!uspot'   # match with uspot option 'setname'

# allow DHCP for captive clients
config rule
        option name 'Allow-DHCP-NTP-captive'
        option src 'captive'
        option proto 'udp'
        option dest_port '67 123'
        option target 'ACCEPT'

# prevent access to LAN-side services from captive interface
# Linux implements a weak host model and traffic crossing zone boundary isn't considered forwarding on the router:
# it must be explicitely denied - NB order matter: DHCP is broadcast that would be caught by this rule
config rule
        option name 'Restrict-input-captive'
        option src 'captive'
        option dest_ip '!captive'
        option target 'DROP'
#       option log '1'

# allow incoming traffic to CPD / web interface and local UAM server
config rule
        option name 'Allow-captive-CPD-WEB-UAM'
        option src 'captive'
        option dest_port '80 443 3990'
        option proto 'tcp'
        option target 'ACCEPT'

# allow forwarding traffic to wan from authenticated clients
config rule
        option name 'Forward-auth-captive'
        option src 'captive'
        option dest 'wan'
        option proto 'any'
        option target 'ACCEPT'
        option ipset 'uspot'    # match with uspot option 'setname'

# allow DNS for captive clients
config rule
        option name 'Allow-DNS-captive'
        option src 'captive'
        list proto 'udp'
        list proto 'tcp'
        option dest_port '53'
        option target 'ACCEPT'

# if using RFC5176 RADIUS DAE:
config rule
        option name 'Allow-captive-DAE'
        option src 'wan'
        option proto 'udp'
        option family 'ipv4'
        option src_ip '1.2.3.4' # adjust as needed
        option dest_port '3799'         # match value for 'das_port' in config/uspot
        option target 'ACCEPT'

# create the ipset that will hold authenticated clients
config ipset
        option name 'uspot'     # match with uspot option 'setname'
        list match 'src_mac'

# optional whitelist for e.g. remote UAM host and/or dynamic hosts via dnsmasq ipset functionality
config rule
        option name 'Allow-Whitelist'
        option src 'captive'
        option dest 'wan'
        option proto 'any'
        option ipset 'wlist'
        option target 'ACCEPT'

# associated whitelist ipset with prepopulated entries
config ipset
        option name 'wlist'
        list match 'dest_ip'
        list entry '35.201.xx.xx'
```

**/etc/config/dhcp**

```
config dhcp 'captive'
        option interface 'captive'
        option start '2'
        option limit '100'
        option leasetime '2h'
        # add the following for RFC8910 Captive Portal API - DNS name is setup below
#       list dhcp_option '114,https://captive.example.org/api'
        # optionally provide NTP server (if enabled on the device) - recommended for SSL cert validation
#       list dhcp_option_force '42,10.0.0.1'

# add a local domain name for HTTPS support, name must match TLS certificate
#config domain
#       option name 'captive.example.org'
#       option ip '10.0.0.1'

# if using optional dynamic hosts whitelist
config ipset
        list name 'wlist'       # match value with whitelist ipset name in config/firewall
        list domain 'mydomain.com'
```

I have also configured a radius dictionary at `/etc/radcli/dictionary`. I note there is also a file at `/etc/radcli/radiusclient.conf` but unsure if that is used for uspot.

From what I can tell, looking at the code, when I enter a username/password on the RADIUS form entry page served by uspot, it gets as far as calling the `/usr/bin/radius_client` binary but then nothing happens - the actual radius call to my configured server does not happen (I'm watching the traffic and nothing to that IP or port 1812 goes out the router)

I can see the following json being generated when it tries to make the above call to `/usr/bin/radius_client`:

```
{ "User-Name": "test", "Calling-Station-Id": "DE:87:26:xx:xx:xx", "Called-Station-Id": "ac:86:74:xx:xx:xx", "Acct-Session-Id": "5af7e59f084099b9", "Framed-IP-Address": "10.0.0.39", "WISPr-Logoff-URL": "http://10.0.0.1:3990/logoff", "Password": "test", "server": "35.230.xx.xx:1812:11111111", "NAS-Identifier": "test" }
```

Log:
```
Mon Apr 15 09:22:21 2024 daemon.info hostapd: phy0-ap0: STA de:87:26:xx:xx:xx IEEE 802.11: authenticated
Mon Apr 15 09:22:21 2024 daemon.info hostapd: phy0-ap0: STA de:87:26:xx:xx:xx IEEE 802.11: associated (aid 1)
Mon Apr 15 09:22:21 2024 daemon.notice hostapd: phy0-ap0: AP-STA-CONNECTED de:87:26:xx:xx:xx auth_alg=open
Mon Apr 15 09:22:21 2024 daemon.info dnsmasq-dhcp[1]: DHCPREQUEST(br-captive) 10.0.0.39 de:87:26:xx:xx:xx
Mon Apr 15 09:22:21 2024 daemon.info dnsmasq-dhcp[1]: DHCPACK(br-captive) 10.0.0.39 de:87:26:xx:xx:xx P7P
Mon Apr 15 09:22:22 2024 daemon.err uhttpd[2262]: uspot: 10.0.0.39 - start radius flow
Mon Apr 15 09:22:25 2024 daemon.err uhttpd[2262]: uspot: 10.0.0.39 - invalid credentials
```

What could be the issue?

Thanks!

**Comment by f00b4r0** [comment]
In all likelihood you haven't added the extra RADIUS dictionaries needed, see [RADIUS-configuration](https://github.com/f00b4r0/uspot/wiki/RADIUS-configuration)

**Comment by PWJW** [comment]
> In all likelihood you haven't added the extra RADIUS dictionaries needed, see [RADIUS-configuration](https://github.com/f00b4r0/uspot/wiki/RADIUS-configuration)

Thanks for the reply!

I thought this too, but as far as I can tell I've done the correct thing...

Initially when I didn't have any dictionary, I received the error:

```
radcli: rc_read_dictionary: rc_read_dictionary couldn't open dictionary /etc/radcli/dictionary: No such file or directory
```

However I added the following file:

**/etc/radcli/dictionary**

```
#
# Updated 97/06/13 to livingston-radius-2.01 miquels@cistron.nl
#
#       This file contains dictionary translations for parsing
#       requests and generating responses.  All transactions are
#       composed of Attribute/Value Pairs.  The value of each attribute
#       is specified as one of 4 data types.  Valid data types are:
#
#       string - 0-253 octets
#       ipv4addr- 4 octets in network byte order
#       ipv6addr  - 16 octets in network byte order
#       ipv6prefix- up to 16 octets in network byte order, plus prefix length
#       integer - 32 bit value in big endian order (high byte first)
#       date - 32 bit value in big endian order - seconds since
#                                       00:00:00 GMT,  Jan.  1,  1970
#
#       Enumerated values are stored in the user file with dictionary
#       VALUE translations for easy administration.
#
#       Example:
#
#       ATTRIBUTE         VALUE
#       ---------------   -----
#       Framed-Protocol = PPP
#       7               = 1     (integer encoding)
#

#
#       Following are the proper new names. Use these.
#
ATTRIBUTE       User-Name               1       string
ATTRIBUTE       Password                2       string
ATTRIBUTE       CHAP-Password           3       string
ATTRIBUTE       NAS-IP-Address          4       ipv4addr
ATTRIBUTE       NAS-Port-Id             5       integer
ATTRIBUTE       Service-Type            6       integer
ATTRIBUTE       Framed-Protocol         7       integer
ATTRIBUTE       Framed-IP-Address       8       ipv4addr
ATTRIBUTE       Framed-IP-Netmask       9       ipv4addr
ATTRIBUTE       Framed-Routing          10      integer
ATTRIBUTE       Filter-Id               11      string
ATTRIBUTE       Framed-MTU              12      integer
ATTRIBUTE       Framed-Compression      13      integer
ATTRIBUTE       Login-IP-Host           14      ipv4addr
ATTRIBUTE       Login-Service           15      integer
ATTRIBUTE       Login-TCP-Port          16      integer
ATTRIBUTE       Reply-Message           18      string
ATTRIBUTE       Callback-Number         19      string
ATTRIBUTE       Callback-Id             20      string
ATTRIBUTE       Framed-Route            22      string
ATTRIBUTE       Framed-IPX-Network      23      ipv4addr
ATTRIBUTE       State                   24      string
ATTRIBUTE       Class                   25      string
ATTRIBUTE       Vendor-Specific         26      string
ATTRIBUTE       Session-Timeout         27      integer
ATTRIBUTE       Idle-Timeout            28      integer
ATTRIBUTE       Termination-Action      29      integer
ATTRIBUTE       Called-Station-Id       30      string
ATTRIBUTE       Calling-Station-Id      31      string
ATTRIBUTE       NAS-Identifier          32      string
ATTRIBUTE       Proxy-State             33      string
ATTRIBUTE       Login-LAT-Service       34      string
ATTRIBUTE       Login-LAT-Node          35      string
ATTRIBUTE       Login-LAT-Group         36      string
ATTRIBUTE       Framed-AppleTalk-Link   37      integer
ATTRIBUTE       Framed-AppleTalk-Network        38      integer
ATTRIBUTE       Framed-AppleTalk-Zone   39      string
ATTRIBUTE       Acct-Status-Type        40      integer
ATTRIBUTE       Acct-Delay-Time         41      integer
ATTRIBUTE       Acct-Input-Octets       42      integer
ATTRIBUTE       Acct-Output-Octets      43      integer
ATTRIBUTE       Acct-Session-Id         44      string
ATTRIBUTE       Acct-Authentic          45      integer
ATTRIBUTE       Acct-Session-Time       46      integer
ATTRIBUTE       Acct-Input-Packets      47      integer
ATTRIBUTE       Acct-Output-Packets     48      integer
ATTRIBUTE       Acct-Terminate-Cause    49      integer
ATTRIBUTE       Acct-Multi-Session-Id   50      string
ATTRIBUTE       Acct-Link-Count         51      integer
ATTRIBUTE       Acct-Input-Gigawords    52      integer
ATTRIBUTE       Acct-Output-Gigawords   53      integer
ATTRIBUTE       Event-Timestamp         55      integer
ATTRIBUTE       Egress-VLANID           56      string
ATTRIBUTE       Ingress-Filters         57      integer
ATTRIBUTE       Egress-VLAN-Name        58      string
ATTRIBUTE       User-Priority-Table     59      string
ATTRIBUTE       CHAP-Challenge          60      string
ATTRIBUTE       NAS-Port-Type           61      integer
ATTRIBUTE       Port-Limit              62      integer
ATTRIBUTE       Login-LAT-Port          63      integer
ATTRIBUTE       Tunnel-Type             64      string
ATTRIBUTE       Tunnel-Medium-Type      65      string
ATTRIBUTE       Tunnel-Client-Endpoint  66      string
ATTRIBUTE       Tunnel-Server-Endpoint  67      string
ATTRIBUTE       Acct-Tunnel-Connection  68      string
ATTRIBUTE       Tunnel-Password         69      string
ATTRIBUTE       ARAP-Password           70      string
ATTRIBUTE       ARAP-Features           71      string
ATTRIBUTE       ARAP-Zone-Access        72      integer
ATTRIBUTE       ARAP-Security           73      integer
ATTRIBUTE       ARAP-Security-Data      74      string
ATTRIBUTE       Password-Retry          75      integer
ATTRIBUTE       Prompt                  76      integer
ATTRIBUTE       Connect-Info            77      string
ATTRIBUTE       Configuration-Token     78      string
ATTRIBUTE       EAP-Message             79      string
ATTRIBUTE       Message-Authenticator   80      string
ATTRIBUTE       Tunnel-Private-Group-ID 81      string
ATTRIBUTE       Tunnel-Assignment-ID    82      string
ATTRIBUTE       Tunnel-Preference       83      string
ATTRIBUTE       ARAP-Challenge-Response 84      string
ATTRIBUTE       Acct-Interim-Interval   85      integer
ATTRIBUTE       Acct-Tunnel-Packets-Lost        86      integer
ATTRIBUTE       NAS-Port-Id-String      87      string
ATTRIBUTE       Framed-Pool             88      string
ATTRIBUTE       Chargeable-User-Identity        89      string
ATTRIBUTE       Tunnel-Client-Auth-ID   90      string
ATTRIBUTE       Tunnel-Server-Auth-ID   91      string
ATTRIBUTE       NAS-Filter-Rule         92      string
ATTRIBUTE       Originating-Line-Info   94      string
ATTRIBUTE       NAS-IPv6-Address        95      ipv6addr
ATTRIBUTE       Framed-Interface-Id     96      string
ATTRIBUTE       Framed-IPv6-Prefix      97      ipv6prefix
ATTRIBUTE       Login-IPv6-Host         98      ipv6addr
ATTRIBUTE       Framed-IPv6-Route       99      string
ATTRIBUTE       Framed-IPv6-Pool        100     string
ATTRIBUTE       Error-Cause             101     integer
ATTRIBUTE       EAP-Key-Name            102     string

#
#       RFC6911 IPv6 attributes
#
ATTRIBUTE       Delegated-IPv6-Prefix   123     ipv6prefix
ATTRIBUTE       Framed-IPv6-Address     168     ipv6addr
ATTRIBUTE       DNS-Server-IPv6-Address 169     ipv6addr
ATTRIBUTE       Route-IPv6-Information  170     ipv6prefix

ATTRIBUTE       Huntgroup-Name          221     string
#
#       Non-Protocol Attributes
#       These attributes are used internally by the server
#
ATTRIBUTE       Expiration                21    date
ATTRIBUTE       Auth-Type               1000    integer
ATTRIBUTE       Menu                    1001    string
ATTRIBUTE       Termination-Menu        1002    string
ATTRIBUTE       Prefix                  1003    string
ATTRIBUTE       Suffix                  1004    string
ATTRIBUTE       Group                   1005    string
ATTRIBUTE       Crypt-Password          1006    string
ATTRIBUTE       Connect-Rate            1007    integer
ATTRIBUTE       User-Category           1029    string
ATTRIBUTE       Group-Name              1030    string
ATTRIBUTE       Simultaneous-Use        1034    integer
ATTRIBUTE       Strip-User-Name         1035    integer
ATTRIBUTE       Fall-Through            1036    integer
ATTRIBUTE       Add-Port-To-IP-Address  1037    integer
ATTRIBUTE       Exec-Program            1038    string
ATTRIBUTE       Exec-Program-Wait       1039    string
ATTRIBUTE       Hint                    1040    string


#
#       Integer Translations
#

#       User Types

VALUE           Service-Type            Login-User              1
VALUE           Service-Type            Framed-User             2
VALUE           Service-Type            Callback-Login-User     3
VALUE           Service-Type            Callback-Framed-User    4
VALUE           Service-Type            Outbound-User           5
VALUE           Service-Type            Administrative-User     6
VALUE           Service-Type            NAS-Prompt-User         7
VALUE           Service-Type            Authenticate-Only       8
VALUE           Service-Type            Callback-NAS-Prompt     9
VALUE           Service-Type            Call-Check              10
VALUE           Service-Type            Callback-Administrative 11

#       Framed Protocols

VALUE           Framed-Protocol         PPP                     1
VALUE           Framed-Protocol         SLIP                    2
VALUE           Framed-Protocol         ARAP                    3
VALUE           Framed-Protocol         GANDALF-SLMLP           4
VALUE           Framed-Protocol         XYLOGICS-IPX-SLIP       5
VALUE           Framed-Protocol         X75                     6

#       Framed Routing Values

VALUE           Framed-Routing          None                    0
VALUE           Framed-Routing          Broadcast               1
VALUE           Framed-Routing          Listen                  2
VALUE           Framed-Routing          Broadcast-Listen        3

#       Framed Compression Types

VALUE           Framed-Compression      None                    0
VALUE           Framed-Compression      Van-Jacobson-TCP-IP     1
VALUE           Framed-Compression      IPX-Header              2
VALUE           Framed-Compression      Stac-LZS                3

#       Login Services

VALUE           Login-Service           Telnet                  0
VALUE           Login-Service           Rlogin                  1
VALUE           Login-Service           TCP-Clear               2
VALUE           Login-Service           PortMaster              3
VALUE           Login-Service           LAT                     4
VALUE           Login-Service           X.25-PAD                5
VALUE           Login-Service           X.25-T3POS              6
VALUE           Login-Service           TCP-Clear-Quiet         8

#       Status Types

VALUE           Acct-Status-Type        Start                   1
VALUE           Acct-Status-Type        Stop                    2
VALUE           Acct-Status-Type        Alive                   3
VALUE           Acct-Status-Type        Accounting-On           7
VALUE           Acct-Status-Type        Accounting-Off          8

#       Authentication Types

VALUE           Acct-Authentic          RADIUS                  1
VALUE           Acct-Authentic          Local                   2
VALUE           Acct-Authentic          Remote                  3

#       Termination Options

VALUE           Termination-Action      Default                 0
VALUE           Termination-Action      RADIUS-Request          1

#       NAS Port Types, available in 3.3.1 and later

VALUE           NAS-Port-Type           Async                   0
VALUE           NAS-Port-Type           Sync                    1
VALUE           NAS-Port-Type           ISDN                    2
VALUE           NAS-Port-Type           ISDN-V120               3
VALUE           NAS-Port-Type           ISDN-V110               4
VALUE           NAS-Port-Type           Virtual                 5
VALUE           NAS-Port-Type           PIAFS                   6
VALUE           NAS-Port-Type           HDLC-Clear-Channel      7
VALUE           NAS-Port-Type           X.25                    8
VALUE           NAS-Port-Type           X.75                    9
VALUE           NAS-Port-Type           G.3-Fax                 10
VALUE           NAS-Port-Type           SDSL                    11
VALUE           NAS-Port-Type           ADSL-CAP                12
VALUE           NAS-Port-Type           ADSL-DMT                13
VALUE           NAS-Port-Type           IDSL                    14
VALUE           NAS-Port-Type           Ethernet                15

#       Acct Terminate Causes, available in 3.3.2 and later

VALUE           Acct-Terminate-Cause    User-Request            1
VALUE           Acct-Terminate-Cause    Lost-Carrier            2
VALUE           Acct-Terminate-Cause    Lost-Service            3
VALUE           Acct-Terminate-Cause    Idle-Timeout            4
VALUE           Acct-Terminate-Cause    Session-Timeout         5
VALUE           Acct-Terminate-Cause    Admin-Reset             6
VALUE           Acct-Terminate-Cause    Admin-Reboot            7
VALUE           Acct-Terminate-Cause    Port-Error              8
VALUE           Acct-Terminate-Cause    NAS-Error               9
VALUE           Acct-Terminate-Cause    NAS-Request             10
VALUE           Acct-Terminate-Cause    NAS-Reboot              11
VALUE           Acct-Terminate-Cause    Port-Unneeded           12
VALUE           Acct-Terminate-Cause    Port-Preempted          13
VALUE           Acct-Terminate-Cause    Port-Suspended          14
VALUE           Acct-Terminate-Cause    Service-Unavailable     15
VALUE           Acct-Terminate-Cause    Callback                16
VALUE           Acct-Terminate-Cause    User-Error              17
VALUE           Acct-Terminate-Cause    Host-Request            18

#
#       Non-Protocol Integer Translations
#

VALUE           Auth-Type               Local                   0
VALUE           Auth-Type               System                  1
VALUE           Auth-Type               SecurID                 2
VALUE           Auth-Type               Crypt-Local             3
VALUE           Auth-Type               Reject                  4

#
#       Cistron extensions
#
VALUE           Auth-Type               Pam                     253
VALUE           Auth-Type               Accept                  254

#
#       Experimental Non-Protocol Integer Translations for Cistron-Radiusd
#
VALUE           Fall-Through            No                      0
VALUE           Fall-Through            Yes                     1
VALUE           Add-Port-To-IP-Address  No                      0
VALUE           Add-Port-To-IP-Address  Yes                     1

#
#       Configuration Values
#       uncomment these two lines to turn account expiration on
#

#VALUE          Server-Config           Password-Expiration     30
#VALUE          Server-Config           Password-Warning        5

$INCLUDE /etc/radcli/dictionary.chillispot
$INCLUDE /etc/radcli/dictionary.WISPr

```

EDIT:
Aha, the last two lines were the issue, including both these from the main dictionary sems to have helped!

Testing now.

Thanks!

**Comment by PWJW** [comment]
Yes, that fixed it!

I have a couple of other questions which I'll raise as a separate topic as not related to this point.

Thanks.
